### DISCUSSION TITLE
Repeater with complex polymorphic relation

### DISCUSSION THREAD

**Discussion by ArtMin96** [filamentphp/filament#11273] [Help]
### Package

Panel builder

### Package Version

v3.2.24

### How can we help you?

I have two types of service table that I need to attach to `App\Models\SomeModel`. But when I set the repeater relation to serviceable, it saves the `serviceable_type` as `App\Models\SomeModel` instead of `App\Models\Services`. And `modelable` columns are not saved at all. I tried as many options as I could think of, but no results. How can I solve this problem?
Thanks

Data need to save like this:
**polymorphic_table**

| serviceable_type    | serviceable_id | modelable_type       | modelable_id | limit |
|---------------------|----------------|----------------------|--------------|-------|
| App\Models\Services | 1              | App\Models\SomeModel | 2            | 10    |

**App\Models\PolymorphicTableModel**
```php
class PolymorphicTableModel extends Model
{
    protected $table = 'polymorphic_table';

    /**
     * @var array<int, string>
     */
    protected $fillable = [
        'serviceable_type',
        'serviceable_id',
        'modelable_type',
        'modelable_id',
        'limit',
    ];

    public function serviceable(): MorphTo
    {
        return $this->morphTo();
    }

    public function modelable(): MorphTo
    {
        return $this->morphTo();
    }
}
```

**App\Models\SomeModel**
```php
class SomeModel extends Model
{
    public function services(): MorphMany
    {
        return $this->morphMany(
            related: PolymorphicTableModel::class,
            name: 'serviceable',
        );
    }

    public function models(): MorphMany
    {
        return $this->morphMany(
            related: SomeModel::class,
            name: 'modelable',
        );
    }
}
```

**App\Models\Services**
```php
class Services extends Model
{
    //...
}
```

**App\Filament\Resources\SomeModelResource**
```php
class SomeModelResource extends Resource
{
    public static function form(Form $form): Form
    {
        return $form
            ->schema([
                Repeater::make('Services')
                    ->relationship('services')
                    ->schema([
                        Select::make('serviceable_id'),
                    ]),
             ]);
    }
}
```

**Comment by micraux** [comment]
I think you made a mistake...

In my opinion, the `models()` function should be in you `App\Models\Services` instead of your `App\Models\SomeModel`, and it should relate to the `PolymorphicTableModel::class` instead of the `SomeModel::class`
